src/greptimeai/extractor/openai_extractor/chat_completion_extractor.py

```diff
@@ -13,15 +13,19 @@
 class ChatCompletionExtractor(OpenaiExtractor):
     def __init__(
         self,
-        client: Optional[OpenAI] = None,
+        client: Optional[Union[OpenAI, AsyncOpenAI]] = None,
         verbose: bool = True,
     ):
+        self.verbose = verbose
+        self.client = client
```

Comment:
we'd better not make client as a field, what about `is_async` field, which is the result of `isinstance(self.client, AsyncOpenAI)`?

and this is a common behavior, what about provide the default override method? and all subclass will parent's `is_async` method?